src/codegen/codegen_c_host.cc

```diff
@@ -37,6 +37,7 @@ CodeGenCHost::CodeGenCHost() {
 void CodeGenCHost::Init(bool output_ssa) {
   decl_stream << "#include \"tvm/runtime/c_runtime_api.h\"\n";
   decl_stream << "#include \"tvm/runtime/c_backend_api.h\"\n";
+  decl_stream << "#include \"tvm/runtime/micro/utvm_device_lib.h\"\n";
```

Comment:
It might make sense to make this include optional based on whether uTVM is being used. The C codegen could be used for other purposes too (e.g., running directly on the host).

Reply:
Yeah.  This is one thing I wasn't sure about, because we also need to modify `codegen_c.cc` (for the ternary operator changes suggested by Tianqi).  So do we want `micro_tvm` flags in both files?

Reply:
@tqchen would appreciate your suggestion here. I'm not sure if we want to involve runtime-level info in the codegen.

Reply:
If we disable ternary operators(to support min/max), we do not need to add any additional declaration headers for now.  

In the future, for other extern functions that needs to be declared. We should have a way to automatically extract a function signature from the call arguments, and write the declaration to decl_stream, without having to include additional header files.

Reply:
The source file being compiled don't need to contain the include, instead, during build, we use the command

```c
gcc -o test_lib.o  micro_tvm_generated.cc src/runtime/micro/utvm_device_lib.cc
```
Where  ```src/runtime/micro/utvm_device_lib.cc``` contains the definition of these functions

Reply:
To be clear, you're suggesting we duplicate the definitions for every created library?

Reply:
It looks like this would pose a problem with checking device errors, because `MicroSession` only has access to the symbol table for the µTVM runtime.  Currently, we just read from the [last_error](https://github.com/dmlc/tvm/pull/3227/files/1791a7b5d00c2501683bc430e326c737e3c1c11c#diff-f1c1afaf885ccf923995f9bffd07bfdcR49) global variable in the runtime, because the session has access to the runtime's symbol table.  But if we now duplicate `last_error` in every loaded binary, the session needs to have access to the location of the `last_error` variable in the module of the function being run.